prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,285 @@
+
+    plt.savefig(get_full_filename("remap_ba_area.tif"))
+    print("BA map completed!")
+
+    return full_ba_map
+
+
+def get_nyiso_area(nydir=get_full_filename("nyiso"), crs=None):
+    """
+    Load the NYISO map from the kmls an geojson into a geo dataframe.
+
+    :param str nydir: NYISO map folder
+    :param  CRS crs: coordinate reference system
+    :return: (*geopandas.DataFrame*)  -- NYISO geoDataFrame
+    """
+    files = os.listdir(nydir)
+    gpd.io.file.fiona.drvsupport.supported_drivers["KML"] = "rw"
+    df_list = []
+    for f in files:
+        if f.endswith("kml"):
+            df = gpd.read_file(os.path.join(nydir, f), driver="KML")
+            df_list.append(df.to_crs(crs))
+        elif f.endswith("geojson"):
+            df = gpd.read_file(os.path.join(nydir, f))
+            df = df.rename(columns={"NAME": "Name"})
+            df = df[["Name", "geometry"]]
+
+            df_list.append(df.to_crs(crs))
+
+    df = pd.concat(
+        df_list,
+    )
+    df_b = df.set_index("Name").loc[["Zone B one", "Zone B two"]].dissolve()
+    df_b["Name"] = "Zone B"
+    df_rest = df.set_index("Name").drop(["Zone B one", "Zone B two"]).reset_index()
+    df = pd.concat([df_b, df_rest])
+    del df["Description"]
+    df = df.rename(columns={"Name": "BA"})
+    col_names = {
+        "Zone A": "NYISO-A",
+        "Zone B": "NYISO-B",
+        "Zone C": "NYISO-C",
+        "Zone D": "NYISO-D",
+        "Zone E": "NYISO-E",
+        "Zone F": "NYISO-F",
+        "Zone G": "NYISO-G",
+        "Millwood": "NYISO-H",
+        "Dunwoodie": "NYISO-I",
+        "Long Island": "NYISO-K",
+        "N.Y.C": "NYISO-J",
+    }
+    df["BA"] = df["BA"].map(col_names)
+    return df.set_index("BA")
+
+
+def download_nyiso_map():
+    """
+    Download the upper NYISO kml file.
+
+    :return: (*None*)
+    """
+    url_dict = {
+        "Zone_A.kml": "https://www.arcgis.com/sharing/rest/content/items/52b8452f80fb4c378a96b382e6c88552/data",
+        "Zone_B_one.kml": "https://www.arcgis.com/sharing/rest/content/items/d0d89f78c77241b482731665e2886941/data",
+        "Zone_B_two.kml": "https://www.arcgis.com/sharing/rest/content/items/1897cf65493a4a76aefba16779df75e8/data",
+        "Zone_C.kml": "https://www.arcgis.com/sharing/rest/content/items/203cf0c4c66c451bb8f220d685bffba8/data",
+        "Zone_D.kml": "https://www.arcgis.com/sharing/rest/content/items/285908f3114547cabb8733ec53883d0b/data",
+        "Zone_E.kml": "https://www.arcgis.com/sharing/rest/content/items/ff3bd2344555499cb1c794a2364eec60/data",
+        "Zone_F.kml": "https://www.arcgis.com/sharing/rest/content/items/19867671655e4db6906a8b7a6b810123/data",
+        "Zone_G.kml": "https://www.arcgis.com/sharing/rest/content/items/a515e6d248db4043b24b9bfd56d0f0ab/data",
+    }
+    for fn in url_dict:
+        response = requests.get(url_dict[fn], stream=True)
+        nyiso2_folder = "nyiso"
+        with open(
+            f"{get_full_filename(os.path.join(nyiso2_folder, fn))}", "wb"
```

Comment:
These should also be changed from `nyiso2` to just `nyiso`.

Reply:
Corrected.